virttest/migration_template.py

```diff
@@ -0,0 +1,673 @@
+            session = None
+            if server_ip == self.remote_ip:
+                server_dict = self.remote_dict
+                session = self.remote_session
+            self.open_port_in_iptables(self.remote_port,
+                                       server_dict=server_dict,
+                                       session=session, cleanup=cleanup)
+
+        logging.debug("Enable libvirtd remote port in firewalld on dst host")
+        _open_libvirtd_port_in_iptables(self.migrate_dest_host, cleanup)
+        if self.migrate_vm_back == 'yes':
+            logging.debug("Enable libvirtd remote port in firewalld\
+                           on src host")
+            _open_libvirtd_port_in_iptables(self.migrate_source_host, cleanup)
+
+    def open_port_in_iptables(self, port, protocol='tcp', server_dict=None,
+                              session=None, cleanup=False):
+        """
+        Open port in iptables
+
+        :param ports: port to be opened, these formats are supported for now:
+                      1)4567 2)4567-4587 3)4567:4587
+        :param protocl: protocol to be opened: 'tcp'|'udp'|'sctp'|'dccp'
+        :param server_dict: dict to create server session:
+                            {server_ip, server_user, server_pwd}
+        :param session: session of server to open ports on, None to local host
+        :param cleanup: True to cleanup instead of opening port
+        """
+        # Check whether firewall-cmd is available
+        use_firewall_cmd = distro.detect().name != "Ubuntu"
+        iptables_func = utils_iptables.Iptables.setup_or_cleanup_iptables_rules
+        try:
+            utils_path.find_command("firewall-cmd")
+        except utils_path.CmdNotFoundError:
+            logging.debug("Using iptables for replacement")
+            use_firewall_cmd = False
+
+        if use_firewall_cmd:
+            firewall_cmd = utils_iptables.Firewall_cmd(session)
+
+            if ":" in port:
+                port = "%s-%s" % (port.split(":")[0], port.split(":")[1])
+
+            # open ports using firewall_cmd
+            if cleanup:
+                firewall_cmd.remove_port(port, protocol, permanent=True)
+            else:
+                firewall_cmd.add_port(port, protocol, permanent=True)
+        else:
+            if "-" in port:
+                port = "%s:%s" % (port.split("-")[0], port.split("-")[1])
+
+            # open migration ports in remote machine using iptables
+            rule = ["INPUT -p %s -m %s --dport %s -j ACCEPT" % (protocol,
+                                                                protocol,
+                                                                port)]
+            iptables_func(rule, params=server_dict, cleanup=cleanup)
+
+        if not cleanup:
+            if session:
+                self.opened_ports_remote.append(port)
+            else:
+                self.opened_ports_local.append(port)
+
+    def cleanup(self):
+        """Cleanup env"""
```

Comment:
please unify the format for docstring with above

Reply:
updated